airbyte-cdk/python/airbyte_cdk/sources/stream_reader/concurrent/stream_partition.py

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Any, Iterable, List, Mapping, Optional
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.stream_reader.concurrent.record import Record
+from airbyte_cdk.sources.streams import Stream
+
+
+class StreamPartition:
+    def __str__(self) -> str:
+        return f"StreamPartition(stream={self._stream.name}, slice={self._slice}, cursor_field={self._cursor_field})"
+
+    def __init__(self, stream: Stream, _slice: Optional[Mapping[str, Any]], cursor_field: Optional[List[str]]):
```

Comment:
Passing the full stream feels heavyhanded. Moving the read logic from the StreamReader to the Stream itself should remove the need to pass the Stream around like this is currently doing